tests/portals/github/test_manager.py

```diff
@@ -0,0 +1,404 @@
+    error = excinfo.value.error
+    assert "Bad credentials" in error.message
+    assert "Github API error:BadCredentialsException" in error.message
+
+
+@pytest.mark.vcr
+def test_github_manager_write_package_file_with_no_credentials(options_write):
+    target_url = options_write.pop("url")
+    with pytest.raises(AssertionError) as excinfo:
+        package = Package("data/datapackage.json")
+        package.to_github(target=target_url, control=portals.GithubControl(apikey=None))
+    assert "AssertionError" in str(excinfo)
+
+
+@pytest.mark.vcr
+def test_github_manager_write_package_file_with_no_params():
+    with pytest.raises(FrictionlessException) as excinfo:
+        package = Package("data/datapackage.json")
+        package.to_github()
+    error = excinfo.value.error
+    assert error.message == "Not supported target: None"
+
+
+@pytest.mark.vcr
+def test_github_manager_write_package_file_with_additional_params(
+    options_write_test_params,
+):
+    target_url = options_write_test_params.pop("url")
+    repo = options_write_test_params.pop("repo")
+    package = Package("data/datapackage.json")
+    response = package.to_github(
+        target=target_url,
+        control=portals.GithubControl(
+            filename="package.json", email="info@okfn.org", name="FD"
+        ),
+    )
+    assert response.repository.url == f"https://api.github.com/repos/fdtester/{repo}"
+    assert repr(response.file_created["content"]) == f'ContentFile(path="package.json")'
+
+
+@pytest.mark.vcr
+def test_github_manager_write_package_file_without_target_url():
+    package = Package("data/datapackage.json")
+    response = package.to_github(
+        control=portals.GithubControl(repo="test-repo-write-without-url")
+    )
+    assert (
+        response.repository.url
+        == f"https://api.github.com/repos/fdtester/test-repo-write-without-url"
+    )
+    assert (
+        repr(response.file_created["content"]) == f'ContentFile(path="datapackage.json")'
+    )
+
+
+# Publish
+
+
+@pytest.mark.vcr
+def test_github_manager_publish_to_github(options_publish_test_params):
+    target_url = options_publish_test_params.pop("url")
+    repo = options_publish_test_params.pop("repo")
+    package = Package("data/datapackage.json")
+    response = package.to_github(target=target_url)
+    assert response.repository.url == f"https://api.github.com/repos/fdtester/{repo}"
+    assert (
+        repr(response.file_created["content"]) == f'ContentFile(path="datapackage.json")'
+    )
+
+
+# Search
+
+
+@pytest.mark.vcr
+def test_github_manager_catalog_from_repo_with_single_package(options_write):
+    repo_url = options_write.pop("url")
+    catalog = Catalog(repo_url)
+
+    # Read
+    expected_file_path = "data/datapackage.json"
+    with open(expected_file_path) as file:
+        assert json.loads(catalog.packages[0].to_json()) == json.loads(file.read())
+
+
+@pytest.mark.vcr
+def test_github_manager_catalog_from_repo_with_multiple_packages(
```

Comment:
@shashigharti 
Let's keep it simple: one repo == one data package so we don't need to create a catalog from one repo